include/swift/AST/SyntaxASTMap.h

```diff
@@ -35,7 +35,7 @@ namespace syntax {
 /// a mapping from lib/AST nodes to lib/Syntax nodes while we integrate
 /// the infrastructure into the compiler.
 class SyntaxASTMap final {
-  friend class LegacyASTTransformer;
+  friend class swift::syntax::LegacyASTTransformer;
```

Comment:
That gives 'C++ requires a type specifier for all declarations' and 'friends can only be classes or functions' when compiling using clang-cl and with `-fms-compatibility-version=19`. I think we're stuck with the explicit namespacing, unfortunately.